### PR TITLE
Add group subgroups data source

### LINKED ISSUE #1177: Group sub-groups, descendent groups, and projects, data sources
### Feature Description

We would like to do some traversing of Gitlab sub-groups with Terraform. In order to be able to do this we would like to see  `gitlab_group_subgroups`, `gitlab_group_descendent`, and `gitlab_group_project` calls added to the provider.

API calls are available for these and they would allow us to manage other resources associated with the Gitlab groups.

### Do you want to implement this?

- [ ] I would like to implement this myself 👷

### PR DESCRIPTION
**Pull request by jcoelho93**
## Description

Partially fixes #1177 
Adds the data source `gitlab_group_subgroups`

### PR Checklist Acknowledgement

- [x] I acknowledge that all of the following items are true, where applicable:
  - Resource attributes match 1:1 the names and structure of the API resource in [the GitLab API documentation](https://docs.gitlab.com/ee/api/).
  - [Examples](https://github.com/gitlabhq/terraform-provider-gitlab/tree/main/examples) are updated with:
    - A \*.tf file for the resource/s with at least one usage example
    - A \*.sh file for the resource/s with an import example (if applicable)
  - New resources have at minimum a basic test with three steps:
    - Create the resource
    - Update the attributes
    - Import the resource
  - No new `//lintignore` comments were copied from existing code. (Linter rules are meant to be enforced on new code.)

### CODE REVIEW

File: internal/provider/resource_gitlab_project.go [hunk 1]
```diff
@@ -1269,12 +1269,7 @@ func resourceGitlabProjectRead(ctx context.Context, d *schema.ResourceData, meta
 func resourceGitlabProjectUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(*gitlab.Client)
 
-	// Always send the name field, to satisfy the requirement of having one
```
Comment (timofurrer) [inline internal/provider/resource_gitlab_project.go:1272]: 🚧 **Suggestion:** Can you please remove these changes in the project resource from this PR?

File: internal/provider/schema_gitlab_group.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
+package provider
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func gitlabGroupSchema() map[string]*schema.Schema {
```
Comment (timofurrer) [inline internal/provider/schema_gitlab_group.go:7]: ❓ **Question:**  what do you think about making use of this in the other group resources / data sources, too?

**Review by timofurrer — CHANGES_REQUESTED** [review]
Looks almost ready to merge 🎉 I've left some comments - back to you 🏓

File: internal/provider/resource_gitlab_project.go (hunk 1 above)
Comment (jcoelho93) [reply internal/provider/resource_gitlab_project.go:1272]: Ups, mix up with branches. Removing

Comment (jcoelho93) [reply internal/provider/resource_gitlab_project.go]: Removed

File: internal/provider/schema_gitlab_group.go (hunk 2 above)
Comment (jcoelho93) [reply internal/provider/schema_gitlab_group.go:10]: I think it makes sense, but we should do that on a different PR, so we don't increase the scope of this one. wdyt?

Comment (jcoelho93) [reply internal/provider/schema_gitlab_group.go:10]: @timofurrer

**Comment by timofurrer** [comment]
Could you please rebase onto the latest main (see #1304)?

**Pull request closed without merging**